src/ucan.ts

```diff
@@ -176,12 +177,33 @@ export function decode(ucan: string): Ucan  {
  * @param ucan The UCAN to encode
  */
 export function encode(ucan: Ucan): string {
-  const encodedHeader = base64.urlEncode(JSON.stringify(ucan.header))
-  const encodedPayload = base64.urlEncode(JSON.stringify(ucan.payload))
+  const encodedHeader = encodeHeader(ucan.header)
+  const encodedPayload = encodePayload(ucan.payload)
 
   return encodedHeader + '.' +
          encodedPayload + '.' +
-         ucan.signature
+         (ucan.signature || sign(ucan.header, ucan.payload))
+}
+
+/**
+ * Encode the header of a UCAN.
+ *
+ * @param header The UcanHeader to encode
+ */
+ export function encodeHeader(header: UcanHeader): string {
+  return base64.urlEncode(JSON.stringify(header))
+}
+
+/**
+ * Encode the payload of a UCAN.
+ *
+ * @param payload The UcanPayload to encode
+ */
+export function encodePayload(payload: UcanPayload): string {
+  return base64.urlEncode(JSON.stringify({
+    ...payload,
+    prf: payload.prf.map(encode)
```

Comment:
Does the server support a list at the moment? 👀

Reply:
@icidasset oh... good catch! server *is* still single proof: https://github.com/fission-suite/fission/blob/main/fission-core/library/Fission/Web/Auth/Token/JWT/Types.hs#L132

thanks!